### PR TITLE
Revisions for help text

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Revises the help text.  This is not [DOC], b/c it's C code, and is TOTALLY UNTESTED; therefore will be marked Draft until tested by reviewers (or, worst case, by me).

### CODE REVIEW

**Comment by nobu** [comment]
Why replacing double quotes with single quotes?
Seems invalid C code.

**Comment by BurdetteLamar** [comment]
> Why replacing double quotes with single quotes? Seems invalid C code.

Wrongthink (Rubyist that I am).  Will fix, and thanks!

**Comment by BurdetteLamar** [comment]
Tests pass, at last!

File: ruby.c [hunk 1]
```diff
@@ -319,80 +319,81 @@ usage(const char *name, int help, int highlight, int columns)
...
-        M("insns_without_opt",      "", "instruction sequences compiled with no optimization"),
-        M("yydebug(+error-tolerant)", "", "yydebug of yacc parser generator"),
-        M("parsetree(+error-tolerant)","", "AST"),
-        M("parsetree_with_comment(+error-tolerant)", "", "AST with comments"),
+        M("insns",                                   "", "Instruction sequences."),
+        M("insns_without_opt",                       "", "Instruction sequences compiled with no optimization."),
+        M("yydebug(+error-tolerant)",                "", "yydebug of yacc parser generator."),
+        M("parsetree(+error-tolerant)",              "", "Abstract syntax tree (AST)."),
+        M("parsetree_with_comment(+error-tolerant)", "", "AST with comments."),
+        M("prism_parsetree",                         "", "Prism AST with comments."),
```
Comment (peterzhu2118) [inline ruby.c:377]: I don't think there are comments included in the dump.

```suggestion
        M("prism_parsetree",                         "", "Prism AST."),
```

File: ruby.c [hunk 2]
```diff
@@ -319,80 +319,81 @@ usage(const char *name, int help, int highlight, int columns)
...
-        M("--jit",         "",                     "enable JIT for the platform, same as " PLATFORM_JIT_OPTION),
+        M("-0[octal]",	   "",                     "Set input record separator ($/):\n"
+            "-0 for \\0; -00 for paragraph mode; -0777 for slurp mode."),
+        M("-a",		   "",                     "Split each input line ($_) into fields ($F)"),
+        M("-c",		   "",			   "Check syntax (no execution)."),
+        M("-Cdirpath",     "",			   "Execute program in specified directory."),
+        M("-d",		   ", --debug",		   "Set debugging flag ($DEBUG) to true."),
+        M("-e 'code'",     "",			   "Execute given ruby code; multiple -e allowed."),
+        M("-Eex[:in]",     ", --encoding=ex[:in]", "Set default internal and external encodings."),
+        M("-Fpattern",	   "",			   "Set input field separator ($;)"),
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply ruby.c]: Fixed.

File: ruby.c [hunk 3]
```diff
@@ -319,80 +319,81 @@ usage(const char *name, int help, int highlight, int columns)
...
+        M("-i[extension]", "",			   "Set ARGF in-place mode."),
+        M("-Idirpath",     "",			   "Add specified directory to load paths ($LOAD_PATH);\n"
+            "multiple -I allowed."),
+        M("-l",		   "",			   "Set output record separator ($\\) to $/"),
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply ruby.c]: Fixed.

File: ruby.c [hunk 4]
```diff
@@ -319,80 +319,81 @@ usage(const char *name, int help, int highlight, int columns)
...
+        M("-n",		   "",			   "Run program in gets loop."),
+        M("-p",		   "",			   "Like -n, with printing added."),
+        M("-rlibrary",	   "",			   "Require the given library."),
+        M("-s",		   "",			   "Define global variables."),
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply ruby.c]: Modified.  Not sure how much we can say here (that'll make sense).

File: ruby.c [hunk 5]
```diff
@@ -319,80 +319,81 @@ usage(const char *name, int help, int highlight, int columns)
...
+        M("-S",		   "",			   "Search directories found in ENV['PATH']."),
+        M("-v",		   "",			   "Print version; set $VERBOSE to true."),
+        M("-w",		   "",			   "Synonym for -W1."),
+        M("-W[lvl=2|:cat]","",   	           "Set warning flag ($-W):\n"
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply ruby.c]: Fixed.  Did not (yet) re-align whitespace in this section, which would make review harder;  can do later if needed.

File: ruby.c [hunk 6]
```diff
@@ -319,80 +319,81 @@ usage(const char *name, int help, int highlight, int columns)
...
+            "0 for silent; 1 for moderate; 2 for verbose."),
+        M("-x[dirpath]",   "",			   "Execute Ruby code found in text."),
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply ruby.c]: Modified.

File: ruby.c [hunk 7]
```diff
@@ -319,80 +319,83 @@ usage(const char *name, int help, int highlight, int columns)
...
+        M("-i[extension]", "",			   "Set ARGF in-place mode;\n"
+            "create backup files with given extension."),
+        M("-Idirpath",     "",			   "Add specified directory to load paths ($LOAD_PATH);\n"
+            "multiple -I allowed."),
+        M("-l",		   "",			   "Set output record separator ($\\) to $/;\n"
+            "used for line-oriented output."),
+        M("-n",		   "",			   "Run program in gets loop."),
+        M("-p",		   "",			   "Like -n, with printing added."),
+        M("-rlibrary",	   "",			   "Require the given library."),
+        M("-s",		   "",			   "Define global variables based on custom options."),
```
Comment (peterzhu2118) [inline ruby.c:340]: I think the original message was more helpful. Maybe something like:

```suggestion
        M("-s",		   "",			   "Define global variables using switches following program path."),
```

File: ruby.c [hunk 8]
```diff
@@ -319,80 +319,83 @@ usage(const char *name, int help, int highlight, int columns)
...
+        M("-S",		   "",			   "Search directories found in the PATH environment variable."),
+        M("-v",		   "",			   "Print version; set $VERBOSE to true."),
+        M("-w",		   "",			   "Synonym for -W1."),
+        M("-W[level=2|:category]", "", 	           "Set warning flag ($-W):\n"
+            "0 for silent; 1 for moderate; 2 for verbose."),
+        M("-x[dirpath]",   "",			   "Execute Ruby code found in text (within certain delimiters)."),
```
Comment (peterzhu2118) [inline ruby.c:346]: Can we be more specific so the user doesn't have to guess what "certain delimiters" is?

```suggestion
        M("-x[dirpath]",   "",			   "Execute Ruby code starting from a #!ruby line."),
```

**Review by peterzhu2118 — APPROVED** [review]

**Comment by BurdetteLamar** [comment]
Thanks, @peterzhu2118.  We'll see this in the next release of Ruby?  Do you know when that will be?

**Comment by peterzhu2118** [comment]
Ruby has a major release every Christmas, so you'll see this in 3.4.0 on December 25.

File: ruby.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (nobu) [reply ruby.c]: The newer text doesn't mention to chdir to `dirpath`.

File: ruby.c [hunk 9]
```diff
@@ -319,80 +319,83 @@ usage(const char *name, int help, int highlight, int columns)
...
-        M("insns_without_opt",      "", "instruction sequences compiled with no optimization"),
-        M("yydebug(+error-tolerant)", "", "yydebug of yacc parser generator"),
-        M("parsetree(+error-tolerant)","", "AST"),
-        M("parsetree_with_comment(+error-tolerant)", "", "AST with comments"),
+        M("insns",                                   "", "Instruction sequences."),
+        M("insns_without_opt",                       "", "Instruction sequences compiled with no optimization."),
+        M("yydebug(+error-tolerant)",                "", "yydebug of yacc parser generator."),
+        M("parsetree(+error-tolerant)",              "", "Abstract syntax tree (AST)."),
+        M("parsetree_with_comment(+error-tolerant)", "", "AST with comments."),
+        M("prism_parsetree",                         "", "Prism AST."),
```
Comment (nobu) [inline ruby.c:379]: `--dump=prism_parsetree` is no longer provided since 839ccad20b9b754261025f65773d1f2178b3d117.
Seems your base revision was too old.

**Comment by BurdetteLamar** [comment]
@nobu, I'm not ignoring your recent comments;  just have some other things to do first.

**Pull request merged**